Casks/e/elemental@6.rb

```diff
@@ -0,0 +1,23 @@
+cask "elemental@6" do
+  version "6.4.0"
+  sha256 "637d1e6fb8de2c36582229cac5b5c6a2ab8b0f735f84ba256efa556f6ecec5fc"
+
+  url "https://github.com/evolvedbinary/elemental/releases/download/elemental-#{version}/elemental-#{version}.dmg",
+      verified: "github.com/evolvedbinary/elemental/"
+  name "elemental"
+  desc "Native XML Database with XQuery and XSLT"
+  homepage "https://www.elemental.xyz/"
+
+  livecheck do
+    url :url
+    regex(/^elemental-(6\..+)$/i)
```

Comment:
```suggestion
      regex(/^elemental[._-]v?(6(?:\.\d+)+)$/i)
```

This regex conforms more closely to our prevailing standards.

Reply:
@bevanjkay Hmm... we actually have pretty strict standards for how our releases are numbered (see: https://github.com/evolvedbinary/elemental/blob/main/VERSIONING_AND_RELEASING.md#versioning-scheme) so the pattern `[._-]v?` would be far too relaxed for us. We also need the ability to have labels and metadata, and so `(?:\.\d+)` would be too restrictive. Does that make sense?

Reply:
I understand what you're saying regarding the version schema, but we use `[._-]v?` across all of our `livecheck` regex, as we tend to err on the side of being more permissive than restrictive with regex, and it also keeps it consistent across the board to reduce the maintenance burden across 1,000s of packages.

Regarding the labels and metadata, based on the versioning and releasing info, a stable release would not have a suffix, so limiting the regex to only semantic versioned tags seem suitable? If I'm misunderstanding this, let me know.

I've also tagged @samford in the PR, because they are our most knowledgable maintainer for `livecheck`.

Reply:
> the pattern `[._-]v?` would be far too relaxed for us

The relative flexibility of this pattern ensures that the regex would continue to match related tags if the format slightly changes from `elemental-1.2.3` to `elemental-v1.2.3`, `elemental_1.2.3`, etc. Unless you're planning to use those variations to differentiate tags in a way that we shouldn't match, there shouldn't be any harm in the added flexibility of this standard pattern. If the version scheme is strictly defined and consistently adhered to, then a more flexible regex should match without issue.

> We also need the ability to have labels and metadata, and so `(?:\.\d+)` would be too restrictive.

Looking through the linked document, it seems like the suffixes primarily apply to unstable versions, which we wouldn't want to match. Are there any stable version formats with a suffix that we would need to handle? If not, then `^elemental[._-]v?(6(?:\.\d+)+)$` would only match stable tags, as expected. `(6\..+)` would match unstable tags like `1.2.3-RC1`, `1.2.3-SNAPSHOT...`, etc., which we don't want (the same applies to `(.+)` in the `elemental` cask).

Reply:
@bevanjkay @samford thank you for your feedback. I have appended a commit which I think should address your suggestions. Please let me know if it needs any further changes...